pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/DistinctCountULLAggregationFunction.java

```diff
@@ -309,7 +309,7 @@ public void aggregateGroupByMV(int length, int[][] groupKeysArray, GroupByResult
   public UltraLogLog extractAggregationResult(AggregationResultHolder aggregationResultHolder) {
     Object result = aggregationResultHolder.getResult();
     if (result == null) {
-      return UltraLogLog.create(_p);
+      return null;
```

Comment:
default ULL object will cause merge failure if p is not the same as what the data has.